cases/api_views.py

```
@@ -252,6 +256,84 @@ def get_queryset(self):
+        if lookup_value.isdigit():
+            # Numeric ID lookup (deprecated)
+            try:
+                obj = queryset.get(id=int(lookup_value))
+
+                # Track deprecation
+                self.request._used_numeric_id = True
+                self._log_numeric_id_deprecation(lookup_value)
+
+                return obj
+            except (Case.DoesNotExist, ValueError):
+                raise Http404("Not found.")
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)